src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionBase.cs

```diff
@@ -185,16 +176,40 @@ private static string ExtractDatabaseNameForLogging(TcpConnectionOptions tcpConn
 
                                     if (await WaitForChangedDocsAsync(state, replyFromClientTask))
                                         continue;
-                                }
-                            }
 
-                            using (batchScope.For(SubscriptionOperationScope.BatchWaitForAcknowledge))
-                            {
-                                replyFromClientTask = await WaitForClientAck(replyFromClientTask);
-                            }
+                                    await CancelSubscriptionAndThrowAsync(replyFromClientTask);
+
+                                    break;
+
+                                case BatchStatus.ActiveMigration:
```

Comment:
you can easily solve this flaw by introducing a method:

`protected virtual bool HandleBatchStatus(...)`

and adding in this base class 2 common cases and in sharded one overriding it and doing:

```
if (status == ActiveMigration)
{
    DoWork();
    return true;
}

return base.HandleBatchStatus();

```

Reply:
done